### PR TITLE
Continuation of TS conversion of components.

### PR DESCRIPTION
**Pull request by celsasser**
# Overview
Committing changes to `components` so that it is up to date. And so that others can get in it and help out.

# Changelog

# Review requests

# Risk assessment

### CODE REVIEW

File: protocol-designer/src/components/BatchEditForm/__tests__/BatchEditMoveLiquid.test.tsx [hunk 1]
```diff
@@ -3,17 +3,17 @@ import React from 'react'
...
 
 describe('BatchEditMoveLiquid', () => {
   const handleCancel = jest.fn()
   const handleSave = jest.fn()
-  let props
+  let props: BatchEditMoveLiquidProps
   beforeEach(() => {
     // just return the i18n text path itself, instead of the text context at that path
-    localizationSpy.mockImplementation(path => path)
+    localizationSpy.mockImplementation(path => path.toString())
```
Comment (shlokamin) [inline protocol-designer/src/components/BatchEditForm/__tests__/BatchEditMoveLiquid.test.tsx]: code change, not a big one but i've been defaulting to throwing in `// @ts-expect-error` statements so preserve testing behavior

File: protocol-designer/src/components/BatchEditForm/__tests__/BatchEditMoveLiquid.test.tsx [hunk 2]
```diff
@@ -43,7 +43,7 @@ describe('BatchEditMoveLiquid', () => {
       expect(localizationSpy).toHaveBeenCalledWith(tooltipPath)
       expect(saveButtonTooltip.prop('children')).toBe(tooltipPath)
 
-      saveButton.invoke('onClick')()
+      saveButton.invoke('onClick')!({} as React.MouseEvent)
```
Comment (shlokamin) [inline protocol-designer/src/components/BatchEditForm/__tests__/BatchEditMoveLiquid.test.tsx]: same here

File: protocol-designer/src/components/BatchEditForm/__tests__/BatchEditMoveLiquid.test.tsx [hunk 3]
```diff
@@ -95,7 +95,7 @@ describe('BatchEditMoveLiquid', () => {
 
       expect(handleCancel).not.toHaveBeenCalled()
 
-      cancelButton.invoke('onClick')()
+      cancelButton.invoke('onClick')!({} as React.MouseEvent)
```
Comment (shlokamin) [inline protocol-designer/src/components/BatchEditForm/__tests__/BatchEditMoveLiquid.test.tsx]: and here

File: protocol-designer/src/components/__tests__/EditModules.test.tsx [hunk 4]
```diff
@@ -71,7 +71,7 @@ describe('Edit Modules', () => {
 
     act(() => {
       editModulesModal.prop('displayModuleWarning')({
-        model: 'some_model',
+        model: 'magneticModuleV1',
```
Comment (shlokamin) [inline protocol-designer/src/components/__tests__/EditModules.test.tsx]: same thing here re:code change. while you are making the type accurate, we are technically changing test behavior. in this case i think it makes sense, but i think the final review will be easier if we preserve what was there before and come back to it later

File: protocol-designer/src/components/BatchEditForm/__tests__/BatchEditMoveLiquid.test.tsx (hunk 1 above)
Comment (celsasser) [reply protocol-designer/src/components/BatchEditForm/__tests__/BatchEditMoveLiquid.test.tsx]: Okay, I'll revise.

**Review by shlokamin — APPROVED** [review]

**Pull request merged**